pkg/pipeline/transform/transform_network.go

```diff
@@ -38,23 +38,14 @@ type Network struct {
 	svcNames *netdb.ServiceNames
 }
 
-func (n *Network) Transform(input []config.GenericMap) []config.GenericMap {
-	outputEntries := make([]config.GenericMap, 0)
-	for _, entry := range input {
-		outputEntry := n.TransformEntry(entry)
-		outputEntries = append(outputEntries, outputEntry)
-	}
-	return outputEntries
-}
-
-func (n *Network) TransformEntry(inputEntry config.GenericMap) config.GenericMap {
+func (n *Network) Transform(inputEntry config.GenericMap) (config.GenericMap, bool) {
 	// copy input entry before transform to avoid alteration on parallel stages
 	outputEntry := inputEntry.Copy()
 
 	// TODO: for efficiency and maintainability, maybe each case in the switch below should be an individual implementation of Transformer
 	for _, rule := range n.Rules {
 		switch rule.Type {
-		case api.TransformNetworkOperationName("AddRegExIf"):
```

Comment:
Logic behind this change is that the result of this invocation is known at runtime while the value of `api.OpAddRegexIf` is known at compile time.

That allows a much more efficient implementation of `switch`, especially important when this is invoked for each flow, for each rule.